Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/ui/consent/one/DccTicketingConsentOneViewModel.kt

```diff
@@ -1,31 +1,138 @@
 package de.rki.coronawarnapp.dccticketing.ui.consent.one
 
+import androidx.lifecycle.LiveData
 import androidx.lifecycle.asLiveData
 import dagger.assisted.Assisted
 import dagger.assisted.AssistedFactory
 import dagger.assisted.AssistedInject
+import de.rki.coronawarnapp.dccticketing.core.common.DccTicketingException
+import de.rki.coronawarnapp.dccticketing.core.service.DccTicketingRequestService
 import de.rki.coronawarnapp.dccticketing.core.transaction.DccTicketingTransactionContext
+import de.rki.coronawarnapp.dccticketing.ui.shared.DccTicketingSharedViewModel
 import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.encryption.ec.EcKeyGenerator
 import de.rki.coronawarnapp.util.ui.SingleLiveEvent
 import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
 import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactory
 import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import timber.log.Timber
 
 class DccTicketingConsentOneViewModel @AssistedInject constructor(
-    @Assisted private val dccTicketingTransactionContext: DccTicketingTransactionContext,
-    dispatcherProvider: DispatcherProvider
+    @Assisted private val dccTicketingSharedViewModel: DccTicketingSharedViewModel,
```

Comment:
Why passing `DccTicketingSharedViewModel` ?  Shared ViewModel is ONLY used to pass data from the main graph to sub-graph due to limitation of passing pracelable when navigating by URI. It is better just to pass the required `data` object. Let us say we found a way to pass the `data` using `args` this approach would be obsolete , but since `DccTicketingConsentOneViewModel` is using `DccTicketingSharedViewModel` it won't be straight foreward. Please keep it the old way and pass only the `data` object from the fragment

Reply:
Another question why do you need  DccTicketingSharedViewModel , don't we have one already to share these data ?